### PR TITLE
Add example how AgileRef can be used instead of AgilePtr

### PR DESCRIPTION
**Pull request by supersega**
Use AgileRef in tests to show how it can be used for marshaling

### CODE REVIEW

File: TutorialsAndTests/Tests/AtlHenTests.cpp [hunk 1]
```diff
@@ -81,22 +84,30 @@ TEST(AtlHenTests, RequireThat_Cluck_IsExecutedOnMainThread_WhenCalledFromWorkerT
...
 
         SetThreadDescription(GetCurrentThread(), L"Worker thread"); // Help debugging
-        HR(agileHen.Get()->CluckAsync(agileObserver.Get()));
+        ComPtr<IHen> henLocal;
+        // All methods in AgileRef calls Resolve to get pointer what is safe to use in this thread.
+        HR(agileHen.As(&henLocal));
+        HR(henLocal->CluckAsync(agileObserver.Get()));
+        // Manually reset ComPtr to avoid deadlock related to destruction after
+        // PostThreadMessage(mainThreadId, WM_QUIT, 0, 0);
+        henLocal.Reset();
```
Comment (jogerh) [inline TutorialsAndTests/Tests/AtlHenTests.cpp:113]: Could you try to figure out why the main thread apartment has to be pumping when henLocal is destroyed? I suspect that henLocal is a proxy, and that when the refcount goes to 0, a message is sent to the main thread to destroy the stub. It should be straight forward to put a breakpoint in the hen destructor to see that it is indeed called from within the message pump? If this is the case, you can maybe make the docstring above explain why we get a deadlock?

**Review by jogerh — CHANGES_REQUESTED** [review]
If you want, the doc string can be improved to explain better what goes on.

File: TutorialsAndTests/Tests/AtlHenTests.cpp (hunk 1 above)
Comment (supersega) [reply TutorialsAndTests/Tests/AtlHenTests.cpp:113]: Yes, it is a destruction of proxy object. I took for granted COM rules, and did paid attention that such string might be helpful.
I will change my PR.

**Review by jogerh — APPROVED** [review]

**Comment by jogerh** [comment]
Good! Thank you @supersega

**Pull request merged**